okhttp/src/main/kotlin/okhttp3/internal/platform/ConscryptPlatform.kt

```diff
@@ -41,26 +43,18 @@ class ConscryptPlatform private constructor() : Platform() {
       SSLContext.getInstance("TLS", provider)
 
   override fun platformTrustManager(): X509TrustManager {
-    return Conscrypt.getDefaultX509TrustManager()
+    val trustManagers = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm()).apply {
+      init(null as KeyStore?)
+    }.trustManagers!!
+    check(trustManagers.size == 1 && trustManagers[0] is X509TrustManager) {
+      "Unexpected default trust managers: ${trustManagers.contentToString()}"
+    }
+    val x509TrustManager = trustManagers[0] as X509TrustManager
+    Conscrypt.setHostnameVerifier(x509TrustManager) { _, _ -> true }
+    return x509TrustManager
   }
 
-  public override fun trustManager(sslSocketFactory: SSLSocketFactory): X509TrustManager? =
-      if (!Conscrypt.isConscrypt(sslSocketFactory)) {
-        super.trustManager(sslSocketFactory)
-      } else {
-        try {
-          // org.conscrypt.SSLParametersImpl
-          val sp = readFieldOrNull(sslSocketFactory, Any::class.java, "sslParameters")
-
-          when {
-            sp != null -> readFieldOrNull(sp, X509TrustManager::class.java, "x509TrustManager")
-            else -> null
-          }
-        } catch (e: Exception) {
-          throw UnsupportedOperationException(
-              "clientBuilder.sslSocketFactory(SSLSocketFactory) not supported on Conscrypt", e)
-        }
-      }
+  override fun trustManager(sslSocketFactory: SSLSocketFactory): X509TrustManager? = null
```

Comment:
Deliberate behaviour change, if a client opts into Conscrypt, they should be knowledgeable enough to provide a trust manager.  FWIW we don't support private use of our internal APIs, so we should stop pushing this upstream also.

Reply:
Yeah, very good idea.